src/Sulu/Bundle/AdminBundle/Resources/js/views/FormOverlayList/tests/FormOverlayList.test.js

```diff
@@ -0,0 +1,384 @@
+// @flow
+import mockReact from 'react';
+import {mount} from 'enzyme';
+import {observable} from 'mobx';
+import FormOverlayList from '../FormOverlayList';
+import List from '../../List';
+import Overlay from '../../../components/Overlay';
+import ResourceStore from '../../../stores/ResourceStore';
+import ResourceFormStore from '../../../containers/Form/stores/ResourceFormStore';
+import Form from '../../../containers/Form';
+import Router from '../../../services/Router';
+import type {Route} from '../../../services/Router';
+import Snackbar from '../../../components/Snackbar';
+
+const React = mockReact;
+
+jest.mock('../../List', () => class ListMock extends mockReact.Component<*> {
+    listStore = {};
+
+    render() {
+        return <div>list view mock</div>;
+    }
+});
+
+jest.mock('../../../containers/Form', () => class ListMock extends mockReact.Component<*> {
+    render() {
+        return <div>form container mock</div>;
+    }
+});
+
+jest.mock('../../../utils/Translator', () => ({
+    translate: jest.fn((key) => key),
+}));
+
+jest.mock('../../../stores/ResourceStore', () => jest.fn(
+    (resourceKey, itemId) => {
+        return {
+            id: itemId,
+        };
+    }
+));
+jest.mock('../../../containers/Form/stores/ResourceFormStore', () => jest.fn(
+    (resourceStore) => {
+        return {
+            id: resourceStore.id,
+        };
+    }
+));
+
+test('View should render with closed overlay', () => {
+    const route: Route = ({}: any);
+    const router: Router = ({
+        route: {
+            options: {},
+        },
+    }: any);
+
+    const formOverlayList = mount(<FormOverlayList route={route} router={router} />);
+
+    expect(formOverlayList.render()).toMatchSnapshot();
+});
+
+test('View should render with opened overlay', () => {
+    const route: Route = ({}: any);
+    const router: Router = ({
+        route: {
+            options: {
+                addOverlayTitle: 'app.add_overlay_title',
+            },
+        },
+    }: any);
+
+    const formOverlayList = mount(<FormOverlayList route={route} router={router} />);
+    formOverlayList.instance().createFormOverlay(undefined);
```

Comment:
Why can't we at least do something like this:

```javscript
formOverlayList.find('List').prop('onItemAdd')();
```

Then we would at least be sure that the function is correctly attached.